File: docs/algorithms/code/algorithms/twoSum2.ts
function twoSum(price: number[], target: number): number[] {
  let left = 0,
    right = price.length - 1;

  while (left < right) {
    if (price[left] + price[right] === target) {
      return [price[left], price[right]];
    }
    if (price[left] + price[right] < target) {
      left++;
      continue;
    }
    if (price[left] + price[right] > target) {
      right--;
      continue;
    }
  }

  return [];
}

File: docs/algorithms/code/algorithms/twoSum2.go
// 思路：采用 hash 表，先记录，后对比
// 复杂度分析：时间复杂度 O(n)，空间复杂度 O(n)
func twoSum(nums []int, target int) []int {
	hash := make(map[int]int)

	for i, v := range nums {
		if p, ok := hash[target-v]; ok {
			return []int{p, i}
		}

		hash[v] = i
	}

	return nil
}